src/helpers/funcs.ts

```diff
@@ -1,3 +1,10 @@
 export const makeStringWithError = (value: string, length: number) => {
     return value.length <= length ? value : `${value.slice(0, length)}<small>${value.slice(length)}</small>`
-}
\ No newline at end of file
+}
+
+
+export const changeRoute = (route: string) => {
+    history.pushState({}, '', "/" + route)
```

Comment:
why make push state twice?

Reply:
It's some kind of hack for react router. First call - we change url but is no redirect to page, second call - we push same adres to history, and then go back one step before for finally page render.
All other ways (like location.href) makes pages reload.

Reply:
you can provide your own wrapper for react router, and encapsulate navigation through it

Reply:
this is more preferable way to do  that, instead of hacks with history API

Reply:
hack is not testable as usual, and very hard to extend it behavior